### ISSUE TITLE
ErrorException: Undefined array key "choices"

### ISSUE THREAD

**Issue by tomirons** [openai-php/client#132]
I'm currently using v0.5.2.

When using the chat completions, I keep getting the following exception when there is a server error on OpenAI's side.

![CleanShot 2023-05-31 at 10 51 45@2x](https://github.com/openai-php/client/assets/5089787/680770d2-2254-4ca1-ada4-31cc71d11530)

While debugging I came across this line.

![CleanShot 2023-05-31 at 10 56 51@2x](https://github.com/openai-php/client/assets/5089787/a88da58e-cbdc-4996-bb03-8ebfc6689273)

Why does it check the charset? Is that required?

If I remove the charset part, then the following exception is thrown which I would expect.

![CleanShot 2023-05-31 at 11 02 52@2x](https://github.com/openai-php/client/assets/5089787/399b7b25-8986-4065-99f1-91d3c2d478e5)

**Comment by gehrisandro** [comment]
Hi @tomirons 

Thank you for raising the issue and your PR.

I made another PR #134  which solves the problem in a more generic way.

Am I right when I assume that the content type is  `application/json` instead of `application/json; charset=utf-8`. Currently I am not able to reproduce the issue by myself.

**Comment by tomirons** [comment]
@gehrisandro yeah, the content type I've been seeing when debugging this is just `application/json`.

**Comment by mi-ke-dev** [comment]
I'm having the same issue today. Thanks for reporting @tomirons.

**Comment by duyemura** [comment]
Hi, I just encountered this when openAI was returning a content_filtered error. 
It would be really nice to know if there was an error on the completion call and then what type of error it was.

To help, here's the completion response on this error:
`CATEGORY:array(1) {
  ["error"]=>
  array(5) {
    ["message"]=>
    string(265) "The response was filtered due to the prompt triggering Azure OpenAI’s content management policy. Please modify your prompt and retry. To learn more about our content filtering policies please read our documentation: https://go.microsoft.com/fwlink/?linkid=2198766"
    ["type"]=>
    NULL
    ["param"]=>
    string(6) "prompt"
    ["code"]=>
    string(14) "content_filter"
    ["status"]=>
    int(400)
  }
}`

**Comment by xpethie** [comment]
I have a similar issue... pls help: got the 
ErrorException
Undefined array key "choices"
in prompt

```
$client = \OpenAI::client(getenv('OPENAI_API_KEY'));
$result_text = $client->completions()->create([
    'model' => 'text-davinci-003',
    'prompt' => 'Analyse this text and give me sentiment of the context on a scale between -1.0 and 1.0 just as decimal without spaces and consider context about Ukraine and Russland or war or death as very negative: ' . $this->article->text,
```

**Comment by gehrisandro** [comment]
This has been fixed in the latest version: 0.5.3

**Comment by mesilov** [comment]
Warning: Undefined array key "choices" in /var/www/html/vendor/openai-php/client/src/Responses/Chat/CreateResponse.php on line 50

Fatal error: Uncaught TypeError: array_map(): Argument #2 ($array) must be of type array, null given in /var/www/html/vendor/openai-php/client/src/Responses/Chat/CreateResponse.php:48

**Comment by iBotPeaches** [comment]
@mesilov - This issue is a few years old and this crash was fixed in 0.16/0.17 so going to lock this to encourage people for a new issue report.
